pkg/vmcp/health/monitor.go

```diff
@@ -0,0 +1,333 @@
+package health
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/stacklok/toolhive/pkg/logger"
+	"github.com/stacklok/toolhive/pkg/vmcp"
+)
+
+// healthCheckContextKey is a marker for health check requests.
+// When present in context, authentication should be bypassed.
+type healthCheckContextKey struct{}
+
+// WithHealthCheckMarker marks a context as a health check request.
+// Authentication layers should skip authentication for these requests.
+func WithHealthCheckMarker(ctx context.Context) context.Context {
+	return context.WithValue(ctx, healthCheckContextKey{}, true)
+}
+
+// IsHealthCheck returns true if the context is marked as a health check.
+func IsHealthCheck(ctx context.Context) bool {
```

Comment:
Is `IsHealthCheck()` being used anywhere? I see it's exported with docs about auth bypass, but couldn't find any auth middleware calling it. If it's for future use, maybe add a note? Otherwise we could make it private.